lib/services/api/api_exception.dart

```diff
@@ -0,0 +1,31 @@
+part of 'api_service.dart';
+
+class ApiException implements Exception {
+  const ApiException({
+    required this.source,
+    required this.message,
+    required this.code,
+  });
+
+  static ApiException? fromHttpException(HttpException exception) {
+    if (exception.type != HttpExceptionType.response &&
+        exception.response?.data is! Map<String, dynamic>) {
+      return null;
+    }
+
+    final json = exception.response!.data as Map<String, dynamic>;
+    final source = json["errors"][0]["source"] as String;
+    final message = json["errors"][0]["detail"] as String;
+    final code = json["errors"][0]["code"] as String;
+
+    return ApiException(
+      source: source,
+      message: message,
+      code: code,
+    );
+  }
+
+  final String source;
+  final String message;
+  final String code;
```

Comment:
better move initial variables on top of the class 🙏

Reply:
I'm following [their style](https://github.com/flutter/flutter/blob/adc687823a/packages/flutter/lib/src/painting/decoration.dart#L24) 💪 

```dart
abstract class Decoration with Diagnosticable {
  /// Abstract const constructor. This constructor enables subclasses to provide
  /// const constructors so that they can be used in const expressions.
  const Decoration();

  @override
  String toStringShort() => objectRuntimeType(this, 'Decoration');

  /// In checked mode, throws an exception if the object is not in a
  /// valid configuration. Otherwise, returns true.
  ///
  /// This is intended to be used as follows:
  /// ```dart
  /// assert(myDecoration.debugAssertIsValid());
  /// ```
  bool debugAssertIsValid() => true;
...
```

Reply:
@markgravity lol, you will be headache because Flutter team has different convention like [this file](https://github.com/flutter/flutter/blob/adc687823a831bbebe28bdccfac1a628ca621513/packages/flutter/lib/src/services/asset_bundle.dart) 🤣 
![image](https://user-images.githubusercontent.com/60863885/116648690-cda97700-a9a7-11eb-8419-feca7ccc0243.png)
And you do not consistent about position of initial variables, please check your `ApiServiceImpl` 🙏

Reply:
@manh-t I don't have any constructor in ApiServiceImpl 🤔 
```dart
class ApiServiceImpl implements ApiService {
  static String? _fallbackBaseUrl;
  static String? _fallbackToken;
  final HttpService _httpService = locator.get();

  @override
  Future<T> call<T extends Mappable>({
  ....
```
Or we don't mention the same section there 😁

Reply:
ah I see 🤔  If you want to follow that style, I'm ok 👌